src/EFCore.PG/Query/Expressions/Internal/FullTextSearchExpression.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Linq;
+using System.Linq.Expressions;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore.Query.Sql.Internal;
+using Microsoft.EntityFrameworkCore.Utilities;
+using NpgsqlTypes;
+
+namespace Microsoft.EntityFrameworkCore.Query.Expressions.Internal
+{
+    public class FullTextSearchExpression : Expression
```

Comment:
Doh! You're right I totally missed that. I guess I was thinking of keeping full text stuff separate and explicit to keep it readable, I will refactor it.

Reply:
Actually for ```@@```, ```@>``` and ```@<``` I'm still forced to create a top-level expression because ```BinaryExpression``` can't be inherited from (its constructors are internal).

Reply:
You're right, I missed that. It's a shame, that would have led to a very clean design.